File: Spring/Chapter4/java_demo/demo/src/main/java/com/aspectj_annotation/AnnotationTest.java
package com.aspectj_annotation;

import org.springframework.context.ApplicationContext;
import org.springframework.context.support.ClassPathXmlApplicationContext;

public class AnnotationTest {
    public static void test()
    {
        ApplicationContext context = new ClassPathXmlApplicationContext("applicationContext.xml");
        AnnotationInterface annotationInterface = (AnnotationInterface)context.getBean("annotationImpl");
        annotationInterface.delete();
        annotationInterface.modify();
        annotationInterface.save();
    }
}

File: Spring/Chapter4/kotlin_demo/demo/src/main/kotlin/aspectj_annotation/AnnotationTest.kt
package aspectj_annotation

import org.springframework.context.ApplicationContext
import org.springframework.context.support.ClassPathXmlApplicationContext

fun test()
{
    val context = ClassPathXmlApplicationContext("applicationContext.xml") as ApplicationContext
    val annotationInterface = context.getBean("annotationImpl") as AnnotationInterface
    annotationInterface.delete()
    annotationInterface.modify()
    annotationInterface.save()
}
